### PR TITLE
Remove `FuelMaterial.__init__()`

### PR DESCRIPTION
**Pull request by keckler**
## Description

`FuelMaterial` has an `__init__()` method which calls directly up to `Material.__init__()`. This makes it difficult to make a new material that has multiple inheritance with, say, `FuelMaterial` and `SimpleSolid`. `FuelMaterial.__init__()` is really not needed anyways, so it was removed in favor of putting the defaults as a class attribute.

This should not impact the user at all.

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by keckler** [comment]
Please hold off on merging this, I might wind back the `puFrac`/`zrFrac`/`uFrac` stuff.

**Comment by keckler** [comment]
Okay, it should be good to merge now, if someone wants to take a look!

**Review by opotowsky — APPROVED** [review]
I don't think I see any issue with this, and kudos on knocking out an issue 🥇

**Comment by keckler** [comment]
This was tested against our internal plugin test suite with no issues.

**Pull request merged**
